lib/hysync/marc_synchronizer/marc_parsing_methods/project.rb

```diff
@@ -10,22 +12,40 @@ module Project
         def add_project(marc_record, holdings_marc_records, mapping_ruleset)
           existing_project = digital_object_data.fetch('project',{})['string_key']
 
-          project_string_key = nil
-          marc_record.fields('965').each do |field|
-            project_string_key = HYSYNC['project_mappings'][field['a']]
-            break unless project_string_key.nil?
-          end
+          project_string_keys = marc_record.fields('965').map do |field|
+            HYSYNC['project_mappings'][field['a']]
+          end.compact
 
-          if project_string_key.nil?
+          if project_string_keys.empty?
             self.errors << "Could not resolve 965 values to a project for: #{self.clio_id}"
             return
           end
-          return if project_string_key.eql?(existing_project)
-          # TODO: In Hyacinth 3, support multiple project associations
-          raise 'This record already has a project.' unless existing_project.nil?
-          digital_object_data['project'] = {
-            'string_key' => project_string_key
-          }
+          # pop the first mapped project off to see if it's the first or actual project for the object
+          first_project_string_key = project_string_keys.shift
+          if existing_project.nil?
+            digital_object_data['project'] = {
+              'string_key' => first_project_string_key
+            }
+          else
+            # if the existing project doesn't equal the first key, put it all in queue for other_project
+            project_string_keys.unshift(first_project_string_key) unless existing_project.eql?(first_project_string_key)
```

Comment:
What you have here is totally fine.  Just noting that I often find myself using `==` for string value comparison instead of `eql?`.  I understand the differences between the comparison, and that they're equivalent in this case, but I don't have a sense of whether the ruby community prefers to use one over the other.

Reply:
Usually you would address this question with a stylesheet 😃  `eql?` is a method I reach for out of habit from my Java days, but it is different from `==` both in human-error potential (assignment typos) and behavior in String: if the argument responds to `to_str` (*not* `to_s`), it is coerced to a String before the comparison.  That's an override - but it conforms to the "looser" sense of `==`; `eql?` is a bit stricter (used for hashing); `equal?` is the strictest (object identity).  In the data implicated here I think it's a matter of style.

Reply:
Ah!  I didn't realize that about `to_str` for string-and-non-string comparisons in `==`.  Good to know!